src/webgl/shaders/cubeFragment.frag

```diff
@@ -0,0 +1,22 @@
+// Adapted from: https://learnopengl.com/PBR/IBL/Diffuse-irradiance
+IN vec3 localPos;
+
+uniform sampler2D equirectangularMap;
+
+const vec2 invAtan = vec2(0.1591, 0.3183);
+
+vec2 SampleSphericalMap(vec3 v)
+{
+    vec2 uv = vec2(atan(v.z, v.x), asin(v.y));
+    uv *= invAtan;
+    uv += 0.5;
+    return uv;
+}
+
+void main()
+{        
+    vec2 uv = SampleSphericalMap(normalize(localPos));
+    vec3 color = texture(equirectangularMap, uv).rgb;
```

Comment:
I feel like, `texture()` is worked on webgl-2 and `texture2D()` is worked on webgl-1. You may use `TEXTURE` to run on both the modes.

Reply:
I had tried that earlier, tried that again today as per your suggestion but both give the same results.
https://github.com/processing/p5.js/pull/6665#discussion_r1461026949
 So i think we should go with `texture` only. :-)

Reply:
hmm...what kind of errors? If you want your code to run on both environments I think `TEXTURE` shouldn't give errors. I don't know why? We can debug it by first writing the code for webgl-1 (changing it to GL ES 100 version I mentioned above, let me know if `texture2D` works there), similarly change it to version 300 and let me know if `texture()` works there? 

Sorry if it seems a long approach for you, But I seriously don't know why `TEXTURE()` is giving you errors. Also, can you share me the screenshots of the errors you are getting.